File: src/main/java/ro/ubb/dp1819/lab3/exercises/pieces/ChessPiece.java
package ro.ubb.dp1819.lab3.exercises.pieces;

public abstract class ChessPiece {

    private Position position;
    private boolean isAlive;
    private ChessPieceColor color;

    public ChessPiece(Position pos, boolean isAlive, ChessPieceColor color) {
        this.position = pos;
        this.isAlive = isAlive;
        this.color = color;
    }

    enum ChessPieceColor{
        WHITE,
        BLACK;
    }

    public Position getPosition() {
        return position;
    }

    public void setPosition(Position position) {
        this.position = position;
    }

    public boolean isAlive() {
        return isAlive;
    }

    public void setAlive(boolean alive) {
        isAlive = alive;
    }

    public ChessPieceColor getColor() {
        return color;
    }

    public void setColor(ChessPieceColor color) {
        this.color = color;
    }
}

File: Laboratoare_gbir2109/DP_Lab3_Homework/DP_Lab3_Homework/Pieces/ChessPiece.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DP_Lab3_Homework.Pieces
{
    public abstract class ChessPiece
    {
        public Position Position { get; set; }

        public bool IsAlive { get; set; }

        public ChessPieceColor Color { get; set; }

        public ChessPiece(int verticalPoz, int horizontalPoz, bool isAlive, ChessPieceColor color)
        {
            this.Position = new Position { HorizontalPos = horizontalPoz, VerticalPos = verticalPoz};
            this.IsAlive = isAlive;
            this.Color = color;
        }
    }

    public enum ChessPieceColor
    {
        WHITE,
        BLACK
    }
}
